transcom/mymove
# INT B-20134 add feature flag to dodid inputs

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20134)
## [Feature Flag Repo PR](https://github.com/transcom/milmove-feature-flags/pull/135#event-12774149177)

## Summary

Utilizing a feature flag to wrap the DODID inputs when pulling data from Okta. This was requested to make testing easier in our testing environments.

This PR does the following:
- adds all relevant feature flag variables to configs
- wraps enablement of the DODID inputs based off of the `okta_dodid_input` feature flag using `useEffect`
- this occurs in the `DodInfoForm.jsx`, `ServiceInfoForm.jsx`, and `index.jsx` in the `OktaInfoFields` folder

### How to test

1. In your `envrc` turn `FEATURE_FLAG_OKTA_DODID_INPUT` to `true`
2. Run `direnv allow` and then `make server_run`
3. Run `make client_run` in another terminal
4. Log in as a customer for the first time and proceed to the DODID form
5. You should see that the DODID input is disabled
6. Set up your profile, create a move, and upload your orders
7. Navigate to the Service Info section of your profile and click Edit
8. You should see the DODID input cannot be updated, but is filled in
9. Navigate to your profile and go to edit your Okta info
10. You should see the DODID input cannot be updated, but is filled in
11. In your `envrc` turn `FEATURE_FLAG_OKTA_DODID_INPUT` to `false`
12. Run `direnv allow` and then `make server_run`
13. Log in as a customer again for the first time and proceed to the DODID form
14. You should now see that the DODID input is enabled
15. Set up your profile, create a move, and upload your orders
16. Navigate to the Service Info section of your profile and click Edit
17. You should see the DODID input CAN be updated
18. Navigate to your profile and go to edit your Okta info
19. You should see the DODID input CAN be updated


## Screenshots

![Screenshot 2024-05-10 at 4 05 29 PM](https://github.com/transcom/mymove/assets/136510600/2e423f47-6b48-48cb-a3df-9c1d915cd657)

![Screenshot 2024-05-10 at 4 05 17 PM](https://github.com/transcom/mymove/assets/136510600/e925e6b9-a474-4256-9452-c1cf6b8ca67b)

![Screenshot 2024-05-10 at 4 05 10 PM](https://github.com/transcom/mymove/assets/136510600/c9e8d918-087a-4ec8-bd8e-7619aa242a6e)
